app/controllers/users_controller.rb

```diff
@@ -0,0 +1,19 @@
+class UsersController < ApplicationController
+
+  def edit_terms
+    if(current_user != nil)
+      user = User.find_by(id: current_user.id)
+      user.update(terms_of_use_accepted: true)
+    end
+
+    if current_user.is_a?(Mentor) && current_user.kids.empty?
+      # if a menotr has no kids yet assigned, go to available kids
+      redirect_to available_kids_path
+    elsif current_user.is_a?(Teacher) && current_user.mentor_matchings.pluck(:state).include?('pending')
```

Comment:
This is a way but the performance could be better-using database query -> using where or exists?

Reply:
Ah I meant something in this manner (using Activerecord) `current_user.mentor_matchings.exists?(state: 'pending')`

Reply:
Ah this. ok. I changed it. And now question - we have same things in application controller after login. Change it too? I just ask because it is not my code but on the other side it is improve.